packages/lexical/src/LexicalSelection.ts

```diff
@@ -1856,11 +1856,11 @@ export class RangeSelection implements BaseSelection {
 
       this.modify('extend', isBackward, 'lineboundary');
 
-      // If selection is extended to cover text edge then extend it one character more
-      // to delete its parent element. Otherwise text content will be deleted but empty
-      // parent node will remain
-      const endPoint = isBackward ? this.focus : this.anchor;
-      if (endPoint.offset === 0) {
+      // If the selection starts at the beginning of a text node (offset 0),
+      // extend the selection by one character in the specified direction.
+      // This ensures that the parent element is deleted along with its content.
+      // Otherwise, only the text content will be deleted, leaving an empty parent node.
+      if (this.anchor.offset === 0) {
```

Comment:
By collapsed I mean that this check should be more like this:

```suggestion
      if (this.isCollapsed() && this.anchor.offset === 0) {
```

Reply:
@etrepum The isCollapsed check had already been made.

https://github.com/facebook/lexical/blob/787b3273023834d4aea44ccdee82405410a8ef8a/packages/lexical/src/LexicalSelection.ts#L1848

Reply:
You're right that a check was already made but the selection is mutated after the check is made with ` this.modify('extend', isBackward, 'lineboundary');` and an `extend` modification (if it does anything) will only change either the anchor or focus thus changing the selection so it's no longer collapsed

Reply:
I see. I've also added a test for the logic of collapsed: [c26567c](https://github.com/facebook/lexical/pull/6719/commits/c26567c6e71a404902590f122703c43d041ee251)